### ISSUE TITLE
feat: Notify of mentions

### ISSUE THREAD

**Issue by nearmax** [near/devgigsboard-widgets#52]
Currently when user's posts is liked, replied to, or edited by a moderator, they receive a notification in near.social UI. However, when they are mentioned, they are not. Mentions are one of the most important use cases, because it allows looping in other people from the outside the conversation.

**Comment by petersalomonsen** [comment]
So this could be implemented by analyzing the post body in the contract and emitting notification events for all account ids that are found? I would suggest in the `add_post` method which is already emitting a notification event for the reply.

**Comment by frol** [comment]
@petersalomonsen yes, that is right. Just for reference, BOS compose component does it on the frontend side: https://near.org/near/widget/ComponentDetailsPage?src=near/widget/Posts.Compose. I think it is fine both ways (do it on the contract side or on the frontend side)

**Comment by petersalomonsen** [comment]
@frol yeah frontend side is maybe safer in terms of not running out of gas. Scanning a large post for mentions might maybe pose a risk of exceeding gas limits, but I guess it has to be tried. An efficient scanning routine might do the job. Also it's probably good to check if the mentioned accounts are valid accounts, and that is probably only possible in the frontend.
